<launch>
  <arg name="robot_ip" default="172.16.0.2"/>
  <arg name="load_gripper" default="true"/>
  <arg name="transmission" default="position"/>

  <arg name="panda_offset_z" default="0.79"/>

  <!-- Hardware -->
  <include file="$(find franka_control)/launch/franka_control.launch">
    <arg name="robot_ip" value="$(arg robot_ip)"/>
    <arg name="load_gripper" value="$(arg load_gripper)"/>
  </include>

  <!-- Controller -->
  <include file="$(find panda_moveit_config)/launch/ros_controllers.launch">
    <arg name="robot_ip" value="$(arg robot_ip)"/>
    <arg name="load_gripper" value="$(arg load_gripper)"/>
    <arg name="transmission" value="$(arg transmission)"/>
  </include>

  <!-- MoveIt ohne Demo -->
  <include file="$(find panda_moveit_config)/launch/move_group.launch"/>

  <!-- Optional RViz -->
  <include file="$(find panda_moveit_config)/launch/moveit_rviz.launch"/>

  <!-- Deine Nodes -->
  <node pkg="sp3_panda_ctrl" type="coordinate_translator" name="coordinate_translator" output="screen"/>

  <node pkg="sp3_panda_ctrl" type="panda_cli_control" name="panda_cli_control" output="screen"
        launch-prefix="gnome-terminal --tab --">
    <param name="panda_offset_z" value="$(arg panda_offset_z)"/>
  </node>
</launch>
<!-- roslaunch sp3_panda_ctrl sp3_real.launch robot_ip:=172.16.0.2 panda_offset_z:=0.805 -->
